### ISSUE TITLE
warning about commondir on ripgrep over git repo with submodules

### ISSUE THREAD

**Issue by Roguelazer** [BurntSushi/ripgrep#1520]
#### What version of ripgrep are you using?

```
$ rg --version
ripgrep 12.0.0
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)
```

Also,

```
$ git --version
git version 2.25.1
```

#### How did you install ripgrep?

Homebrew

#### What operating system are you using ripgrep on?

macOS. Also seems to occur on Linux.

#### Describe your question, feature request, or bug.

When running ripgrep on a git clone with submodules, it prints the following error:

```
 $ rg -i foobar
../../../../.git/modules/path/to/submodule/commondir: No such file or directory (os error 2)
```

I suspect this is related to #1446 

#### If this is a bug, what are the steps to reproduce the behavior?

```
git init foo
git init bar
cd foo
touch file
git add file
git commit -m 'initial commit'
cd ../bar
touch file
git add file
git commit -m 'initial commit'
git submodule add ../foo path
rg string
```

#### If this is a bug, what is the actual behavior?

The ripgrep invocation will print

```
../.git/modules/path/commondir: No such file or directory (os error 2)
```

#### If this is a bug, what is the expected behavior?

No warnings. Never warnings.

**Comment by BurntSushi** [comment]
Oof, sorry. You're right. This is a regression.

I will put out a new release soonish. I'll give it a few days in case any other regressions pop up.

**Comment by joshuarubin** [comment]
Any chance you could cut a release now? This issue is driving me batty!

**Comment by BurntSushi** [comment]
I'd recommend just building from source if it's that bad. Otherwise: https://github.com/BurntSushi/ripgrep/blob/master/FAQ.md#release

**Comment by BurntSushi** [comment]
I've released ripgrep 12.0.1 with a fix for this.
